tests/treatment_effect/matching/test_propensity.py

```diff
@@ -0,0 +1,116 @@
+"""Tests for the PropensityMatching class in the matching module."""
+
+from __future__ import annotations
+
+import unittest
+from typing import TYPE_CHECKING, List, Optional, Set
+
+import pandas as pd
+import pytest
+
+from medmodels import MedRecord
+from medmodels.treatment_effect.matching.propensity import PropensityMatching
+
+if TYPE_CHECKING:
+    from medmodels.medrecord.types import NodeIndex
+
+
+def create_patients(patients_list: List[NodeIndex]) -> pd.DataFrame:
+    """Creates a patients dataframe.
+
+    Args:
+        patients_list (List[NodeIndex]): List of patients to include in the dataframe.
+
+    Returns:
+        pd.DataFrame: A patients dataframe.
+    """
+    patients = pd.DataFrame(
+        {
+            "index": ["P1", "P2", "P3", "P4", "P5", "P6", "P7", "P8", "P9"],
+            "age": [20, 30, 40, 30, 40, 50, 60, 70, 80],
+            "gender": [
+                "male",
+                "male",
+                "female",
+                "female",
+                "male",
+                "male",
+                "female",
+                "female",
+                "male",
+            ],
+        }
+    )
+
+    return patients.loc[patients["index"].isin(patients_list)]
+
+
+def create_medrecord(patients_list: Optional[List[NodeIndex]] = None) -> MedRecord:
+    """Creates a MedRecord object.
+
+    Args:
+        patients_list (Optional[List[NodeIndex]], optional): List of patients to include
+            in the MedRecord. Defaults to None.
+
+    Returns:
+        MedRecord: A MedRecord object.
+    """
+    if patients_list is None:
+        patients_list = [
+            "P1",
+            "P2",
+            "P3",
+            "P4",
+            "P5",
+            "P6",
+            "P7",
+            "P8",
+            "P9",
+        ]
+    patients = create_patients(patients_list=patients_list)
+    medrecord = MedRecord.from_pandas(nodes=[(patients, "index")])
+    medrecord.add_group(group="patients", nodes=patients["index"].to_list())
+    medrecord.add_nodes(("P10", {}), "patients")
+    return medrecord
```

Comment:
these two functions seem to be duplicated at least once. You can put them in  a `helper.py` file and use them from there.

Reply:
also applies to other helper functions that are duplicated